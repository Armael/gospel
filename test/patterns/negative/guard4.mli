type t = A

val f : t -> int
(*@ r = f x
  requires match x with
           | A when 1 = 1 -> true
           | _ when true -> false
*)
