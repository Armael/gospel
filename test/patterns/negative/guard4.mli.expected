
*******************************
********** Parsed file ********
*******************************
type t =
  | A 
val f : t -> int[@@gospel
                  {| r = f x
  requires match x with
           | A when 1 = 1 -> true
           | _ when true -> false
|}]

*******************************
****** GOSPEL translation *****
*******************************
(*@ open Stdlib *)

(*@ open Gospelstdlib *)

type t =
| A 
  

val f : t -> int
(*@ r = f x
    requires ...
     *)
File "guard4.mli", line 5, characters 11-91:
Error: This pattern-matching may not be exhaustive because of the guard.
       Here is an example of a case that may not be matched:
         A.
